ddtrace/contrib/starlette/patch.py

```diff
@@ -184,3 +193,27 @@ async def traced_task(*args, **kwargs):
 
     args, kwargs = set_argument_value(args, kwargs, 0, "func", traced_task)
     wrapped(*args, **kwargs)
+
+
+def _fix_span_url(scope, request_spans):
+    """
+    Fix the URL tag on the request span for Starlette versions <= 0.33.0
+    """
+
+    root_path = scope.get("root_path", "")
+
+    # Only run if there are child spans that need a valid root to be appended
+    if not request_spans and len(request_spans) > 1 and root_path:
```

Comment:
How about the scenario where you have multiple nested mounts. For example:

```
app = FastAPI()
subapp1 = FastAPI()
subapp2 = FastAPI()

@subapp1.get("/hello1/{name}")
def hello1():
    return {"Greeting1": "Hello"}

@subapp2.get("/hello2/{name}")
def hello2():
    return {"Greeting2": "Hello"}

app1.mount("/sub-app2", subapp2)
app.mount("/sub-app1", subapp1)
```

It seems like the scope used to set the root_path in `sub-app2` be equal to `sup-app1` (which is incorrect).  We should add this as an additional snapshot test and verify it works.